letsencrypt/client/client.py

```diff
@@ -336,7 +336,8 @@ def init_csr(privkey, names, cert_dir):
     csr_pem, csr_der = crypto_util.make_csr(privkey.pem, names)
 
     # Save CSR
-    le_util.make_or_verify_dir(cert_dir, 0o755)
+    if not os.environ.get('DOCKER_RUN'):
```

Comment:
As @jdkasten and I discussed, this is due to a permissions error that emerges when the host path is shared with the container.  We circumvent this by setting this `DOCKER_RUN` env var in the `Dockerfile`, then only checking permissions at runtime if it is not set.